src/kernel/task.zig

```diff
@@ -254,3 +290,39 @@ test "allocatePid and freePid" {
 
     expectEqual(all_pids.bitmap, 0);
 }
+
```

Comment:
Could have a test to go over all the errors to ensure the memory cleanup.
#224 😉

Reply:
If you mean heap memory then I will add an extra `testing.expectError` call for each error that the function can return. If you mean VMM memory, then I have been thinking about that and would like to implement VMM memory leak detection like the testing allocator has, but I think that is outside of the scope of this PR.